### PR TITLE
[API] User can create a group

### PR DESCRIPTION
**Pull request by Jon1969Edwards**
## Description
As a User
In order to create a community of people with same interest
I would like to be able to create a group

## PivotalTracker
[PT Link](https://www.pivotaltracker.com/story/show/163513648)

## Tasks
- Create request specs
- Create Group #create action endpoint
- Set up Group#create action

### CODE REVIEW

**Review by CamronLDNF — APPROVED** [review]
Looks good to me. Ping Faraz and Thomas to also review and subsequently merge it for us.

File: app/controllers/groups_controller.rb [hunk 1]
```diff
@@ -6,4 +6,17 @@ def show
     group = Group.find(params[:id])
     render json: group, serializer: Groups::ShowSerializer
   end
-end
\ No newline at end of file
+
+  def create
+    group = Group.new(params.require(:group).permit(:name))
```
Comment (tochman) [inline app/controllers/groups_controller.rb]: This is not following convention. We've been sanitizing params in a separate method before.

File: spec/requests/user_can_create_group_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    it 'responds with success message' do
+      expect(response_json['message']).to eq 'You have created a group successfully'
+    end
+
+    it 'responds with status 200' do
+      expect(response).to have_http_status 200
+    end
+  end
+end
```
Comment (tochman) [inline spec/requests/user_can_create_group_spec.rb:56]: You need to add a spec for failing to create a group

File: app/controllers/groups_controller.rb (hunk 1 above)
Comment (Jon1969Edwards) [reply app/controllers/groups_controller.rb]: Right, ... We've refactored this now.

File: spec/requests/user_can_create_group_spec.rb (hunk 2 above)
Comment (Jon1969Edwards) [reply spec/requests/user_can_create_group_spec.rb:56]: Of course.  Has been addressed and fixed.

**Pull request merged**